src/main/java/uk/gov/justice/digital/hmpps/keyworker/services/StaffConfigManager.kt

```diff
@@ -58,6 +64,86 @@ class StaffConfigManager(
     }
   }
 
+  fun patchStaffConfiguration(
+    prisonCode: String,
+    staffId: Long,
+    request: PatchStaffConfigRequest,
+  ) {
+    val staffConfig = staffConfigRepository.findByStaffId(staffId) ?: throw EntityNotFoundException("Staff config not found")
+    if (!request.jobClassification.isPresent || request.jobClassification.get() != null) {
+      staffConfig.patch(request)
+    }
+
+    if (request.deactivateActiveAllocations.orElse(false) ||
+      (request.jobClassification.isPresent && request.jobClassification.get() == null)
+    ) {
+      deallocateAllocationsFor(prisonCode, staffId)
+    }
+
+    request.jobClassification.ifPresent {
+      if (it == null) {
+        deactivateJobRole(prisonCode, staffId)
+        staffConfigRepository.deleteByStaffId(staffId)
+      } else {
+        patchJobRole(prisonCode, staffId, it)
+      }
+    }
+  }
+
+  private fun deactivateJobRole(
+    prisonCode: String,
+    staffId: Long,
+  ) = AllocationContext.get().policy.nomisUseRoleCode?.let {
+    prisonApi.getKeyworkerForPrison(prisonCode, staffId)?.apply {
+      nurApi.setStaffRole(
+        prisonCode,
+        staffId,
+        it,
+        StaffJobClassificationRequest(
+          position = position,
+          scheduleType = scheduleType,
+          hoursPerWeek = hoursPerWeek,
+          fromDate = fromDate,
+          toDate = LocalDate.now(),
+        ),
+      )
+    } ?: throw EntityNotFoundException("Staff role not found")
```

Comment:
For idempotency purposes, it's fine if it can't find the existing job classification. Means it's already deactivated. This shouldn't throw an exception